### PR TITLE
Gather mem utilization

### PR DESCRIPTION
**Pull request by roznawsk**

### CODE REVIEW

File: lib/beamchmark/formatters/html/templates.ex [hunk 1]
```diff
@@ -98,6 +98,23 @@ defmodule Beamchmark.Formatters.HTML.Templates do
...
+
+    %{
+      memory_usage:
+        Enum.map_join(memory_snapshots, ", ", fn %{
+                                                   total: total_bytes
+                                                 } ->
+          total_bytes
+        end),
+      time: Enum.map_join(1..length(memory_snapshots), ", ", fn el -> el end)
+    }
```
Comment (DominikWolek) [inline lib/beamchmark/formatters/html/templates.ex]:
```suggestion
    %{
      memory_usage:
        Enum.map_join(memory_snapshots, ", ", fn %{total: total_bytes} -> total_bytes end),
      time: Enum.map_join(1..length(memory_snapshots), ", ", fn el -> el end)
    }
```

File: lib/beamchmark/suite/measurements/memory_info.ex [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    average =
+      Enum.reduce(mem_types, %{}, fn mem_type, average ->
+        mem_type_avg =
+          Enum.reduce(memory_snapshots, 0, fn snapshot, sum ->
+            sum + Map.get(snapshot, mem_type)
+          end)
+          |> div(length(memory_snapshots))
+
+        Map.put(average, mem_type, mem_type_avg)
+      end)
```
Comment (DominikWolek) [inline lib/beamchmark/suite/measurements/memory_info.ex]: I think this is more readable and cleaner:
```suggestion
    average =
      Enum.reduce(mem_types, %{}, fn mem_type, average ->
        mem_type_avg =
          memory_snapshots
          |> Enum.map(&Map.get(&1, mem_type))
          |> Enum.sum()
          |> div(length(memory_snapshots))

        Map.put(average, mem_type, mem_type_avg)
      end)
```

File: lib/beamchmark/suite/memory/memory_task.ex [hunk 3]
```diff
@@ -0,0 +1,38 @@
+defmodule Beamchmark.Suite.Memory.MemoryTask do
+  @moduledoc """
+  This module contains the memory benchmarking task.
+  Measurements are performed using [`:erlang.memory/0`](https://www.erlang.org/doc/man/erlang.html#memory-0)
+  Run example:
+  ```
+  MemoryTask.start_link()
+  ```
```
Comment (DominikWolek) [inline lib/beamchmark/suite/memory/memory_task.ex]: I don't really get this "run example", both here and in `CpuTask`:

- `start_link` takes 2 arguments, not 0.
- They explain very basic thing, meanwhile we have more sophisticated examples in the `examples` folder.

File: lib/beamchmark/suite/memory/memory_task.ex [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  @spec run_poll(number(), number()) :: {:ok, MemoryInfo.t()}
+  defp run_poll(mem_interval, duration) do
+    iterations_number = trunc(duration / mem_interval)
+
+    memory_snapshots =
+      Enum.reduce(0..(iterations_number - 1), [], fn _x, memory_snapshots ->
+        memory_snapshots = [memory_snapshot() | memory_snapshots]
+        Process.sleep(mem_interval)
+        memory_snapshots
+      end)
```
Comment (DominikWolek) [inline lib/beamchmark/suite/memory/memory_task.ex]:
```suggestion
    memory_snapshots =
      Enum.map(0..(iterations_number - 1), fn _x ->
        Process.sleep(mem_interval)
        memory_snapshot()
      end)
```

File: lib/beamchmark/suite/memory/memory_task.ex (hunk 3 above)
Comment (roznawsk) [reply lib/beamchmark/suite/memory/memory_task.ex]: I removed those examples.

**Review by DominikWolek — APPROVED** [review]

**Pull request merged**